arbnode/inbox_reader.go

```diff
@@ -77,6 +93,12 @@ type InboxReader struct {
 }
 
 func NewInboxReader(tracker *InboxTracker, client arbutil.L1Interface, l1Reader *headerreader.HeaderReader, firstMessageBlock *big.Int, delayedBridge *DelayedBridge, sequencerInbox *SequencerInbox, config *InboxReaderConfig) (*InboxReader, error) {
+	if config.MaxBlocksToRead == 0 || config.MaxBlocksToRead < config.DefaultBlocksToRead {
+		return nil, errors.New("inbox reader max-blocks-to-read cannot be zero or less than default-blocks-to-read")
+	}
+	if config.AdjustBlocksToRead && config.TargetMessagesRead == 0 {
+		return nil, errors.New("inbox reader target-messages-read cannot be zero if adjust-blocks-to-read is true")
+	}
```

Comment:
we should have the config validation in a separate function.. we'll probably want to use this when we implement live config updates